### DISCUSSION TITLE
Send temperature and humidity values from i2c-sensor to KNX

### DISCUSSION THREAD

**Discussion by empty88** [arendst/Tasmota#15539] [Ideas]
hello everybody,
i connected a si7021 to my sonoff basic. The temperature and humidity values are shown on the webinterface as expected and are sent over mqtt successfully. Now I wanted to use KNX but the temperature and humidity options in the knx settings are not visible.
![image](https://user-images.githubusercontent.com/29315283/166808836-0b6cd1d7-491e-4082-a893-62fbdb746f47.png)

I have to select the one-wire si7021 for one gpio, that is not in use to see the options.
![image](https://user-images.githubusercontent.com/29315283/166808656-f3ec2f16-fb5a-493a-a2fa-0ac8e571b0e8.png)

With this workaround I can use KNX. but I think it is not the best way.
![image](https://user-images.githubusercontent.com/29315283/166809135-036f4bf6-ab28-41cf-8f9a-f4cfffe1c97c.png)

**Comment by ascillato** [comment]
Hi,

I can not reproduce your issue. Sorry.

Please, can you try with the latest precompiled release version of KNX?

http://ota.tasmota.com/tasmota/release/tasmota-knx.bin.gz

Or try to do a `reset 6` in the console and configure your device again.

**Reply by empty88** [reply]
Hi,

the precompiled version cannot work because USE_HTU is not set.
[Firmware builds](https://tasmota.github.io/docs/Firmware-Builds/#available-features-and-sensors)

You need USE_HTU for the si7021 to be recognized over i2c.
`#define USE_HTU            // [I2cDriver9] Enable HTU21/SI7013/SI7020/SI7021 sensor (I2C address 0x40) (+1k5 code)`

**Reply by ascillato** [reply]
You are right. The SI7021 is in the KNX version but SI7021 I2C is not. I was testing with a simple SI7021. I will add the support for the SI7021 I2C in the KNX driver.

Changing this to a feature request. Stay tuned. Thanks for noticing this.

**Reply by empty88** [reply]
Cool! Thanks!
